From 4db20110ee7664ab51760de8f508bbb4abe785b6 Mon Sep 17 00:00:00 2001
From: Tyler Gunn <tgunn@google.com>
Date: Mon, 8 Apr 2019 13:34:41 -0700
Subject: [PATCH 6/8] Add missing null check on start of
 SipIncomingCallReceiver received.

Test: Ran ADB command to start receiver; verified no crash.
Fixes: 129903757
Change-Id: I89f912eedeb70958a59f24e09075fedfe52dca14
---
 .../services/telephony/sip/SipIncomingCallReceiver.java     | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sip/src/com/android/services/telephony/sip/SipIncomingCallReceiver.java b/sip/src/com/android/services/telephony/sip/SipIncomingCallReceiver.java
index c754ae442..19f58825f 100644
--- a/sip/src/com/android/services/telephony/sip/SipIncomingCallReceiver.java
+++ b/sip/src/com/android/services/telephony/sip/SipIncomingCallReceiver.java
@@ -25,6 +25,7 @@ import android.os.UserHandle;
 import android.telecom.PhoneAccount;
 import android.telecom.PhoneAccountHandle;
 import android.telecom.TelecomManager;
+import android.text.TextUtils;
 import android.util.Log;
 
 /**
@@ -38,6 +39,11 @@ public class SipIncomingCallReceiver extends BroadcastReceiver {
     public void onReceive(Context context, final Intent intent) {
         String action = intent.getAction();
 
+        if (TextUtils.isEmpty(action)) {
+            log("SipIncomingCallReceiver called with no action");
+            return;
+        }
+
         if (!isRunningInSystemUser()) {
             if (VERBOSE) log("SipIncomingCallReceiver only run in system user, ignore " + action);
             return;
-- 
2.17.1

